### PR TITLE
README: rework sections: zanzibar, contrib, users

### PR DESCRIPTION
**Pull request by jzelinskie**
This reworks the README a little bit
- Adds a high-level architecture image to "What is" section (works in dark mode too!)
- Reworks text on "What is" section to lean on relational databases and mention relation to authN
- Adds a Zanzibar section
- Adds a "Who's using" section with a few interesting community projects; we can workshop these over time, but i aimed to get some variety in domains and languages
- Moved reading materials from contributing to Getting Started
- Added some contributor companies in the contributing section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,13 +33,34 @@
 </p>
 
 ## What is SpiceDB?
+<a href="https://authzed.com#gh-dark-mode-only" target="_blank"><img align="right" width="300" src="https://github.com/user-attachments/assets/e0e70549-91dc-4a07-9309-2e18942a7902" alt="spicedb diagram" /></a>
+<a href="https://authzed.com#gh-light-mode-only" target="_blank"><img align="right" width="300" src="https://github.com/user-attachments/assets/97342fce-d94b-42b6-b02f-fe3958d13981" alt="spicedb diagram" /></a>
```
Comment (tstirrat15) [inline README.md:37]: nice

File: README.md [hunk 2]
```diff
@@ -221,6 +256,6 @@ We appreciate all contributions, large and small, and would like to thank all th
...
-- <img alt="netflix logo" height="15px" src="https://github.com/authzed/spicedb/assets/343539/e64128f0-978f-4fd6-bdd7-1ce7cb6b34b9"> The Netflix Authorization Team for sponsoring and being a design partner for caveats
-- <img alt="equinix logo" height="15px" src="https://github.com/authzed/spicedb/assets/343539/7bf706f9-910d-4902-8957-c914a7468eff"> The Equinix Metal Team for sponsoring our benchmarking hardware
+- <a href="#"><img alt="github logo" height="15px" src="https://github.com/authzed/spicedb/assets/343539/c05b8aef-c862-4499-bebf-0a43f3b423c4"></a> The GitHub Authorization Team for implementing and contributing the MySQL datastore
+- <a href="#"><img alt="netflix logo" height="15px" src="https://github.com/authzed/spicedb/assets/343539/e64128f0-978f-4fd6-bdd7-1ce7cb6b34b9"></a> The Netflix Authorization Team for sponsoring and being a design partner for caveats
+- <a href="#"><img alt="equinix logo" height="15px" src="https://github.com/authzed/spicedb/assets/343539/7bf706f9-910d-4902-8957-c914a7468eff"></a> The Equinix Metal Team for sponsoring our benchmarking hardware
```
Comment (tstirrat15) [inline README.md:261]: Curious - why wrap?

**Review by tstirrat15 — APPROVED** [review]
Looks great to me

File: README.md (hunk 2 above)
Comment (jzelinskie) [reply README.md:261]: put this in the PR description:
- Wrapped images in a anchor tag to prevent github from turning them into links

**Review by tstirrat15 — APPROVED** [review]
LGTM

**Pull request merged**